packages/extension-ui/src/components/View.tsx

```diff
@@ -18,9 +20,10 @@ interface Props {
 function View({ children, className }: Props): React.ReactElement<Props> {
   const [theme, setTheme] = useState(chooseTheme());
 
-  const switchTheme = useCallback((theme: AvailableThemes): void => {
-    localStorage.setItem('theme', theme);
+  const switchTheme = useCallback(async (theme: AvailableThemes): Promise<void> => {
     setTheme(theme);
+
+    await localStorageStores.theme.set(theme);
```

Comment:
I reversed the order so that the components rerender immediately, without waiting for the local storage commit. I thought that in the likely scenario of something failing in between, the theme value in storage is not that important anyway.

Reply:
one possible issue could be that someone queries local storage on re-render and might get value inconsistent with `theme` - it's unlikely, but on the other hand I don't think a delay from awaiting local storage should be that significant :thinking:

Reply:
Hmm yeah that's true.. Not sure how React handles rendering now with all this new, interruptive Fiber engine, but I think that pre-Fiber the case you describe might actually be quite common, since `setState` is synchronous. So with Fiber it's probably gonna be more random-ish. I think you've convinced me to swap it back.

Reply:
> since setState is synchronous

agh, this doesn't have anything to do with async storage reading.. but I'm swapping back anyway